sdk/attestation/azure-security-attestation/src/attestation_client.cpp

```diff
@@ -215,11 +211,68 @@ void AttestationClient::RetrieveResponseValidationCollateral(
   }
 }
 
-void AttestationClient::CheckAttestationSigners() const
+/** @brief Construct a new Attestation Client object
+ *
+ * @param endpoint The URL address where the client will send the requests to.
+ * @param credential The authentication method to use (required for TPM attestation).
+ * @param options The options to customize the client behavior.
+ */
+AttestationClient const AttestationClient::Create(
+    std::string const& endpoint,
+    std::shared_ptr<Core::Credentials::TokenCredential const> credential,
+    AttestationClientOptions options,
+    Azure::Core::Context const& context)
 {
-  std::unique_lock<std::shared_timed_mutex> stateLock(SharedStateLock);
+  AttestationClient returnValue(endpoint, credential, options);
+  returnValue.RetrieveResponseValidationCollateral(context);
+  return returnValue;
+}
+
+/** @brief Construct a new anonymous Attestation Client object
+ *
+ * @param endpoint The URL address where the client will send the requests to.
+ * @param options The options to customize the client behavior.
+ *
+ * @note TPM attestation requires an authenticated attestation client.
+ */
+AttestationClient const AttestationClient::Create(
+    std::string const& endpoint,
+    AttestationClientOptions options,
+    Azure::Core::Context const& context)
+{
+  return Create(endpoint, nullptr, options, context);
+}
+
+/** @brief Construct a new Attestation Client object
+ *
+ * @param endpoint The URL address where the client will send the requests to.
+ * @param credential The authentication method to use (required for TPM attestation).
+ * @param options The options to customize the client behavior.
+ */
+AttestationClient* const AttestationClient::CreatePointer(
+    std::string const& endpoint,
+    std::shared_ptr<Core::Credentials::TokenCredential const> credential,
+    AttestationClientOptions options,
+    Azure::Core::Context const& context)
+{
+  std::unique_ptr<AttestationClient> returnValue(
+      new AttestationClient(endpoint, credential, options));
+  returnValue->RetrieveResponseValidationCollateral(context);
+  // Release the client pointer from the unique pointer to let the parent manage it.
+  return returnValue.release();
```

Comment:
```suggestion
  auto returnValue = std::make_unique<AttestationClient>(endpoint, credential, options);
  returnValue->RetrieveResponseValidationCollateral(context);
  return returnValue;
```

Reply:
Ah,  I was missing the constructor is now private.  Then the issue is not the std::unique_ptr, but calling `std::make_unique<T>` (It has not access to private).

So, instead of calling `std::make_unique<T>`, you can just do:
```cpp
static std::unique_ptr<Factory> Create() {
        auto instance = std::unique_ptr<Factory>(new Factory());
        return instance;
    }
```
Then you can return the unique_ptr.  See this https://godbolt.org/z/obs3b3zae

Reply:
To my eyes, that is almost identical to the original code with the exception of removing the .release, which I removed earlier.  What is the difference here?